### PR TITLE
Upgrade Resiliency: Refactors Implementation for Opening Rntbd Connections to Backend Replicas in Direct Mode.

### LINKED ISSUE #3336: Parameterize server ssl cert name override in direct connectivity mode
**Context** - In scenarios where the client service cannot take a dependency on Dns (i.e connecting to cosmosDb server using IP).
The ssl handshake with direct connectivity fails since url doesn't match the SN/SAN of the server certificate.

**Ask** - Can we parameterize "servercerthostnameoverride" (probably with clientoptions.cs or cosmosclientbuilder) so that it can be appropriately passed to documentclient while creating the storeclient.

https://github.com/Azure/azure-cosmos-dotnet-v3/blob/63734436ff87925d2612c5bfce19b815e25811d0/Microsoft.Azure.Cosmos/src/DocumentClient.cs#L6533

### LINKED ISSUE #3547: [Cosmos Service Upgrade - .NET] Replica Validation Azure.Cosmos.Direct package changes
- Implement code changes in `Azure.Cosmos.Direct` package for replica validation.

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

Refactors the V3 SDK code to use the latest interfaces included in `Cosmos.Direct` `v3.30.0` to open and initialize Rntbd channels to the backend replica nodes, in Direct connectivity mode. Additionally it includes the Direct package changes for parameterizing server ssl cert name override in **Direct** connectivity mode.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Closing issues

To automatically close an issue: 
closes #3547
closes #3336

### CODE REVIEW

**Review by ealsur — DISMISSED** [review]

**Review by aavasthy — APPROVED** [review]
👍

File: Directory.Build.props [hunk 1]
```diff
@@ -4,7 +4,7 @@
 		<ClientOfficialVersion>3.31.2</ClientOfficialVersion>
 		<ClientPreviewVersion>3.31.2</ClientPreviewVersion>
 		<ClientPreviewSuffixVersion>preview</ClientPreviewSuffixVersion>
-		<DirectVersion>3.29.4</DirectVersion>
+		<DirectVersion>3.30.0</DirectVersion>
```
Comment (ealsur) [inline Directory.Build.props:7]: As part of the Direct bump, add Poland Central as part of the `Regions.cs`

```
        /// <summary>
        /// Name of the Azure Poland Central region in the Azure Cosmos DB service.
        /// </summary>
        public const string PolandCentral = "Poland Central";
```

Do a quick verification that creating a client with ApplicationRegion = Regions.PolandCentral does not throw any exceptions

Comment (kundadebdatta) [reply Directory.Build.props:7]: Validated that Cosmos Client initialization doesn't throw exception for `PolandCentral` application region.

**Review by ealsur — DISMISSED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (ealsur) [reply Directory.Build.props:7]: Probably requires Contract changes

Comment (kundadebdatta) [reply Directory.Build.props:7]: As discussed, I have reverted the changes for this particular PR.

File: Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs [hunk 2]
```diff
@@ -88,7 +92,7 @@ internal class GatewayAddressCache : IAddressCache, IDisposable
             string databaseName,
             ContainerProperties collection,
             IReadOnlyList<PartitionKeyRangeIdentity> partitionKeyRangeIdentities,
-            Func<Uri, Task> openConnectionHandler,
+            bool shouldOpenRntbdChannels,
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs:95]: Please code comment on when is it expected to be off/on.

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs (hunk 2 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs:95]: Sure, taken a note of it and will address it with the replica validation PR. Tx!

**Pull request merged**